feature-staking-impl/src/main/java/io/novafoundation/nova/feature_staking_impl/domain/nominationPools/main/unbondings/NominationPoolUnbondingsInteractor.kt

```diff
@@ -38,15 +38,14 @@ class RealNominationPoolUnbondingsInteractor(
         sharedComputationScope: CoroutineScope,
     ): Flow<Unbondings> {
         val chainId = stakingOption.assetWithChain.chain.id
-        return combineTransform(
+        return combineToPair(
             stakingSharedComputation.activeEraFlow(chainId, sharedComputationScope),
             nominationPoolSharedComputation.unbondingPoolsFlow(poolMember.poolId, chainId, sharedComputationScope),
-        ) { activeEraIndex, unbondingPools ->
-            val unbondingsFlow = unbondingPools.unbondingsFor(poolMember, activeEraIndex, stakingOption, sharedComputationScope)
-                .map { Unbondings.from(it, rebondPossible = false) }
-
-            emitAll(unbondingsFlow)
-        }
+        )
+            .flatMapLatest { (activeEraIndex, unbondingPools) ->
+                unbondingPools.unbondingsFor(poolMember, activeEraIndex, stakingOption, sharedComputationScope)
+            }
+            .map { Unbondings.from(it, rebondPossible = false) }
```

Comment:
Explanations for expected unstake fix:

When using `emitAll` inside `combineTranform`, next pair of values wont be processed untill `emitAll` finishes emitting values, which is never happens since since ` unbondingPools.unbondingsFor` is an infinite flow

To demostrate `combineTrasnform + emitAll` behavior lets use the following example:



``` kotlin
val aFlow = flow {
            (1..100).map {
                println("A${it}")
                emit(it)

                kotlinx.coroutines.delay(500)
            }
        }

        val bFlow = flow {
            (1..100).map {
                println("B${it}")
                emit(it)


                kotlinx.coroutines.delay(500)
            }
        }

        combineTransform(aFlow, bFlow) { a, b ->
            val c = flow {
                (1..10).map {
                    println("A${a}:B${b} - $it")
                    emit(it)

                    kotlinx.coroutines.delay(200)
                }
            }

            emitAll(c)
        }.collect()
```

Result will be the following:

```
A1
B1
A1:B1 - 1
A1:B1 - 2
A1:B1 - 3
A2
B2
A1:B1 - 4
A1:B1 - 5
A3
B3
A3:B2 - 1
```

Reply:
Thank you for explanation! It's easier to understand now